src/hooks/use-subscription-data.tsx

```diff
@@ -1,27 +1,39 @@
 import * as React from 'react'
 import axios from 'axios'
 
+type SubscriptionData = {
+  portalUrl?: string
+  billingScheme: 'tiered' | 'per_unit'
+  subscription?: any
+  price?: any
+  product?: any
+  latestInvoice?: any
+  upcomingInvoice?: any
+}
+
+export const recur = (price: any) => {
+  if (price === undefined) return ''
+
+  const {
+    recurring: {interval, interval_count},
+  } = price
+
+  if (interval === 'month' && interval_count === 3) return 'quarter'
+  if (interval === 'month' && interval_count === 6) return '6-months'
+  if (interval === 'month' && interval_count === 1) return 'month'
+  if (interval === 'year' && interval_count === 1) return 'year'
+}
```

Comment:
Exporting recur instead of passing it around as a prop since it is a static concept.